### ISSUE TITLE
[Bug]: OpenTTD

### ISSUE THREAD

**Issue by szentigrad3** [parkervcp/eggs#1874]
### Panel Version

1.10.1

### Wings Version

1.7.0

### Service

OpenTTD

### Modified

No, I did not modify the egg

### Expected Behavior

when you 1st start the server would expect it to generate a new world, and then after load a save

### Actual Behavior

generates a new map everytime i restart the server
https://i.imgur.com/EULlQMN.png

### Steps To Reproduce

N/A

### Install logs

N/A

**Comment by QuintenQVD0** [comment]
is this set tot true?

https://github.com/parkervcp/eggs/blob/e27be233427da478983376936ba45db01513b3e2/game_eggs/tycoon_games/openttd/openttd.cfg#L314

**Comment by szentigrad3** [comment]
> is this set tot true?
> 
> https://github.com/parkervcp/eggs/blob/e27be233427da478983376936ba45db01513b3e2/game_eggs/tycoon_games/openttd/openttd.cfg#L314

it was not, testing it now.. also keep getting this error > https://i.imgur.com/hWuvxrI.png

**Comment by szentigrad3** [comment]
> > is this set tot true?
> > https://github.com/parkervcp/eggs/blob/e27be233427da478983376936ba45db01513b3e2/game_eggs/tycoon_games/openttd/openttd.cfg#L314
> 
> it was not, testing it now.. also keep getting this error > https://i.imgur.com/hWuvxrI.png

testing did not work, just keeps making a new map.. lost

**Comment by szentigrad3** [comment]
i need ls -t save/autosave | head -n1 | xargs -I {} cp -f save/autosave/{} save/autosave/latest_autosave.sav to be pre_start bebfore running the startup command ./openttd -D -g  {{SAVEGAME}}

**Comment by szentigrad3** [comment]
made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`

then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server

**Comment by szentigrad3** [comment]
[egg-open-t-t-d-server.json.txt](https://github.com/parkervcp/eggs/files/9648704/egg-open-t-t-d-server.json.txt)
needs more testing

**Comment by QuintenQVD0** [comment]
> made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`
> 
> then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server

That is because your in the proces of the .sh file. Hava a look at bash onliners multiple commands with a delay between them

**Comment by szentigrad3** [comment]
> > made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`
> > then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server
> 
> That is because your in the proces of the .sh file. Hava a look at bash onliners multiple commands with a delay between them

too dumb for that part

**Comment by szentigrad3** [comment]
> > made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`
> > then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server
> 
> That is because your in the proces of the .sh file. Hava a look at bash onliners multiple commands with a delay between them

any chance you can finish this up for me please, i did have a look at bash onliners multiple commands with a delay between them... but way over my head

**Comment by QuintenQVD0** [comment]
> > > made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`
> > > then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server
> > 
> > 
> > That is because your in the proces of the .sh file. Hava a look at bash onliners multiple commands with a delay between them
> 
> any chance you can finish this up for me please, i did have a look at bash onliners multiple commands with a delay between them... but way over my head

I can but not rn. The whole week I am bussy. Mayby next monday

**Comment by szentigrad3** [comment]
> > > > made a start.sh `[ -f save/autosave/latest_autosave.sav ] && ls -t save/autosave | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav || ./openttd -D`
> > > > then you also have to change Start Configuration cause it changes the startup.... but now exit doesn't work to stop the server
> > > 
> > > 
> > > That is because your in the proces of the .sh file. Hava a look at bash onliners multiple commands with a delay between them
> > 
> > 
> > any chance you can finish this up for me please, i did have a look at bash onliners multiple commands with a delay between them... but way over my head
> 
> I can but not rn. The whole week I am bussy. Mayby next monday

can wait until then, thanks

**Comment by parkervcp** [comment]
It doesn't look like openttd has a function to load a save on start. You need to load the save from the console https://wiki.openttd.org/en/Manual/Console#game-commands

**Comment by szentigrad3** [comment]
> It doesn't look like openttd has a function to load a save on start. You need to load the save from the console https://wiki.openttd.org/en/Manual/Console#game-commands

./openttd -D -g save/autosave/latest_autosave.sav the -g option loads a save on startup

**Comment by parkervcp** [comment]
the issue I see here then is that there is no specific "latest" autosave?

**Comment by szentigrad3** [comment]
> the issue I see here then is that there is no specific "latest" autosave?

I made a script that if there is no sav it will start a new map then if.you have a save it will find the newest auto save and rename it to that. The problem is now when using the script the stop server command doesn't work

**Comment by parkervcp** [comment]
Yeah that's because it's a script. I might have a better way to do it where it "should" pull the latest autosave by date.

**Comment by szentigrad3** [comment]
> Yeah that's because it's a script. I might have a better way to do it where it "should" pull the latest autosave by date.

this is the script have atm

#!/bin/bash
shopt -s nullglob
sav_files=(save/autosave/*.sav)
if ((${#sav_files[@]}))
then
  ls -t "save/autosave" | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav
else
  ./openttd -D
fi

**Comment by szentigrad3** [comment]
any updates on this?

**Comment by QuintenQVD0** [comment]
Currently no

**Comment by parkervcp** [comment]
I have been poking at different ways to have it get the latest

You can have it save on exit and that will create an `exit.sav` but that won't always be the latest file in cases of crashes.

**Comment by szentigrad3** [comment]
> I have been poking at different ways to have it get the latest
> 
> You can have it save on exit and that will create an `exit.sav` but that won't always be the latest file in cases of crashes.

this is why i have this script it grabs the latest one and renames it >
#!/bin/bash
shopt -s nullglob
sav_files=(save/autosave/*.sav)
if ((${#sav_files[@]}))
then
ls -t "save/autosave" | head -n1 | xargs -I {} cp -uf save/autosave/{} save/autosave/latest_autosave.sav | ./openttd -D -g save/autosave/latest_autosave.sav
else
./openttd -D
fi
